cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CommandUtil.java

```diff
@@ -207,6 +247,394 @@ public static void applyTemplate(Path modulePath, String template) throws IOExce
+            }
+        }
+
+        Path balaPath = balaCache.resolve(
+                ProjectUtils.getRelativeBalaPath(orgName, packageName, version, null));
+
+        String platform = findPlatform(balaPath);
+        balaPath = balaCache.resolve(
+                ProjectUtils.getRelativeBalaPath(orgName, packageName, version, platform));
+
+        Gson gson = new Gson();
+        String packageJsonPath = String.valueOf(balaPath.resolve("package.json"));
+        try (FileReader packageReader = new FileReader(packageJsonPath)) {
+            PackageJson packageJson = gson.fromJson(packageReader, PackageJson.class);
+            boolean templateVal = packageJson.getTemplate();
+            if (Files.exists(balaPath.resolve("package.json"))) {
+                if (templateVal) {
+                    userDir = Paths.get(System.getProperty(ProjectConstants.USER_DIR));
+                    // Copy docs
+                    Path packageMDFilePath = balaPath.resolve("docs")
+                            .resolve(ProjectConstants.PACKAGE_MD_FILE_NAME);
+                    Path toPackageMdPath = modulePath.resolve(ProjectConstants.PACKAGE_MD_FILE_NAME);
+                    if (Files.exists(packageMDFilePath)) {
+                        Files.copy(packageMDFilePath, toPackageMdPath, StandardCopyOption.REPLACE_EXISTING);
+                    }
+
+                    Path moduleMd = balaPath.resolve("docs").resolve("modules").resolve(packageName)
+                            .resolve(ProjectConstants.MODULE_MD_FILE_NAME);
+                    Path toModuleMd = modulePath.resolve(ProjectConstants.MODULE_MD_FILE_NAME);
+                    if (Files.exists(moduleMd)) {
+                        Files.copy(moduleMd, toModuleMd);
+                    }
+
+                    // Copy modules
+                    Path sourceModulesDir = balaPath.resolve("modules").resolve(packageName);
+                    if (Files.exists(sourceModulesDir)) {
+                        Files.walkFileTree(sourceModulesDir, new FileUtils.Copy(sourceModulesDir, modulePath));
+                    }
+
+                    // Copy platform library
+                    Path platformLibPath = balaPath.resolve("platform").resolve("java11");
+                    Path projectPlatform = modulePath.resolve("Platform");
+                    Path platformLibRelativePath = userDir.relativize(projectPlatform);
+                    if (Files.exists(platformLibPath)) {
+                        createDirectories(projectPlatform);
+                        Files.walkFileTree(platformLibPath, new FileUtils.Copy(platformLibPath, projectPlatform));
+                    }
+
+                    // Copy Package.json to Ballerina.toml
+                    Path balaToml = modulePath.resolve(ProjectConstants.BALLERINA_TOML);
+                    Files.createFile(balaToml);
+                    writeBallerinaToml(balaToml, packageJson, platformLibRelativePath, orgName, packageName,
```

Comment:
Why have we done the same thing as the method `applyBalaTemplate` here? Looks like this has some duplicated content. Can you please check?